### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by otpas007** [wzhouad/ATLOP#7]
Hi,
I was trying to train a custom dataset with the ATLOP based on the bert-base-cased model. But I found that the dev-f1 is stuck ~28-29 and not increasing even after training 30 epochs. I also checked the train loss graph but it is showing the normal behaviour (decreasing throughout and reaching order of 10^-2 in the end). 
Can you please suggest how to train the model to achieve better dev-F1 score?

The train_loss and dev-f1 score graphs are attached herewith for reference.

Train-loss vs steps
![Train-loss_vs_steps](https://user-images.githubusercontent.com/76780827/103615888-6b88ba00-4f51-11eb-83a7-7deced6c6690.PNG)

Dev-F1 vs steps
![Dev-F1_vs_steps](https://user-images.githubusercontent.com/76780827/103615907-72173180-4f51-11eb-988c-0ce6f0706673.PNG)

**Comment by wzhouad** [comment]
Hi,

As no enough information is provided, I don't know the exact cause of poor performance, but I suggest to check the following parts of your code:
1. Is the training behavior normal? Decrease in training loss does not necessarily indicate better performance. You can output the training F1 curve to ensure the code working properly.
2. Is the label mapping correct? In the implementation of the adaptive thresholding loss I fix the NA class to an index of 0. You need to map the NA class to index 0 if there's an NA class in your new dataset. If there's no NA class, you can change the loss function to standard cross entropy loss.
